gem2rpm/smart_proxy_plugin.spec.erb

```diff
@@ -5,6 +5,9 @@ root = `git rev-parse --show-toplevel`.strip
 SCL_PREFIXES = JSON.load(File.read(File.join(root, 'scl_prefixes.json')))
 LICENSES = JSON.load(File.read(File.join(root, 'licenses.json')))
 has_cronjob = spec.files.any? { |f| f =~ %r{\Aextra/.*\.cron\z } }
+
+config.rules[:ignore] << 'bundler.d'
+config.rules[:ignore] << 'settings.d'
```

Comment:
Rather than ignoring them, how about adding `rmdir` in `%install`. That way we know for sure we don't accidentally miss files when they're added (since the build will fail).

@evgeni?

Reply:
I am not sure that I follow the problem here.

An entry in the ignore list essentially generates an `%exclude`, right?

And `rmdir` would only work if the directory is *empty*, which it isn't, the plugin dropped its example config there?

Reply:
In theory they would be empty at the end of the `%install` section, because we perform a `mv` to copy both locations:

e.g. https://github.com/theforeman/foreman-packaging/pull/6497/files#diff-2fd47368d60a1da78f86557b3824e35a393f46a8ce13b5f596a8367064f93d02R96-R97

Reply:
What @ehelms said indeed. If we're not moving all files, we may be missing something. The `rmdir` should catch that.

Reply:
Okay, the move was what was missing from my view.

Yeah, then `rmdir` sounds nice

Reply:
This gets a bit more complicated. If I add the `rmdir` this does not prevent it from showing in the files list everytime you run gem2rpm. The macro used (https://github.com/theforeman/foreman-packaging/blob/rpm/develop/gem2rpm/smart_proxy_plugin.spec.erb#L165) takes as input the files being reported from the gemspec not what exists in the buildroot. So, I would have to add a rmdir *and* update this chunk of code to explicitly ignore `bundler.d` and `settings.d` directories. That feels like more work than it should be when I can use built in ignores to just exclude directories.

Thoughts?

Reply:
Ironically, Pulp is the only plugin that I know of that used to ship multiple config files. Especially Pulp would benefit from `rmdir`. I still think that both is the right solution.

Also, there's code to support `config/` in plugins in the same way as `settings.d/` so I think it should be
```suggestion
config.rules[:ignore] << 'bundler.d'
config.rules[:ignore] << 'settings.d'
config.rules[:ignore] << 'config'
```

Reply:
I'm a bit confused by what the `rmdir` would accomplish. If you run `rmdir` inside `%install` section you are essentially just removing it from the buildroot which has no effect on the final set of files available and laid down by the RPM itself.

Reply:
```console
$ mkdir example
$ touch example/file
$ rmdir example
rmdir: failed to remove 'example': Directory not empty
```

So if you add a file to settings.d but don't update the spec file, the mv won't be there and thus rmdir should fail. This means the build fails which is a good thing. Otherwise it would silently not work and only once you start testing you find out something is missing.

That is my reasoning. You get an early failure which makes iterating on changes faster.

Reply:
Now that makes sense, appreciate the explanation.

Reply:
And to complete the round trip of this, the example case that you are looking ahead to is accounted for already via the way the template is written (as I can parse it).

https://github.com/theforeman/foreman-packaging/blob/rpm/develop/gem2rpm/smart_proxy_plugin.spec.erb#L136-L146

This detects all files in `settings.d/` or `config` and generates statements to ensure they get moved, for example, this plugin when it had Pulp 2 support, the following is generated:

```
mkdir -p %{buildroot}%{foreman_proxy_settingsd_dir}
mv %{buildroot}%{gem_instdir}/settings.d/pulp.yml.example \
   %{buildroot}%{foreman_proxy_settingsd_dir}/pulp.yml
mv %{buildroot}%{gem_instdir}/settings.d/pulpcore.yml.example \
   %{buildroot}%{foreman_proxy_settingsd_dir}/pulpcore.yml
mv %{buildroot}%{gem_instdir}/settings.d/pulpnode.yml.example \
   %{buildroot}%{foreman_proxy_settingsd_dir}/pulpnode.yml
```

Two of those lines are removed in this PR because as part of updating it the spec file also got updated to match the new state of the gem.

Reply:
If there is more to do here, could I move this change to an issue that is stand-alone (since in theory we'd want to apply this change to all smart-proxy plugins) to get this particular update of the smart_proxy_pulp plugin built and allow me to additionally merge https://github.com/theforeman/puppet-foreman_proxy/pull/638 as those two things will effectively close out the removal of Pulp 2 from the installer.

Reply:
I am happy with the current changes